crates/common/crypto/keccak/mod.rs

```diff
@@ -0,0 +1,367 @@
+#[cfg(target_arch = "aarch64")]
+std::arch::global_asm!(include_str!("keccak1600-armv8.s"), options(raw));
+#[cfg(target_arch = "x86_64")]
+std::arch::global_asm!(include_str!("keccak1600-x86_64.s"), options(att_syntax));
+
+pub use imp::*;
+
+#[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))]
+mod imp {
+    const BLOCK_SIZE: usize = 136;
+
+    #[derive(Default, Clone, Copy)]
+    #[repr(transparent)]
+    struct State([u64; 25]);
+
+    unsafe extern "C" {
+        #[link_name = "SHA3_absorb"]
+        unsafe fn SHA3_absorb(state: *mut State, buf: *const u8, len: usize, r: usize) -> usize;
+        unsafe fn SHA3_squeeze(state: *mut State, buf: *mut u8, len: usize, r: usize);
+    }
+
+    pub fn keccak_hash(data: impl AsRef<[u8]>) -> [u8; 32] {
+        let mut state = Keccak256::new();
+        state.update(data);
+        state.finalize()
+    }
+
+    #[derive(Clone)]
+    pub struct Keccak256 {
+        state: State,
+        tail_buf: [u8; BLOCK_SIZE],
+        tail_len: usize,
+    }
+
+    impl Default for Keccak256 {
+        fn default() -> Self {
+            Self {
+                state: State::default(),
+                tail_buf: [0; BLOCK_SIZE],
+                tail_len: 0,
+            }
+        }
+    }
```

Comment:
This can be derived in the declaration of the struct. All the fields are taking their default value.